### ISSUE TITLE
Infinite loop with Layout/SpaceInsideArrayLiteralBrackets and Layout/SpaceInsideReferenceBrackets

### ISSUE THREAD

**Issue by aquach** [bbatsov/rubocop#5289]
## Expected behavior

Here's the configuration:

```
Layout/SpaceInsideArrayLiteralBrackets:
  EnforcedStyle: space
```

And here's the code:

```
a = []
a[[ 1, 2 ]]
```

I expect this to pass. SpaceInsideArrayLiteralBrackets should be happy since there is a space inside the brackets. SpaceInsideReferenceBrackets should be happy since there are no spaces in between the `[[` and `]]`. It should complain if I were doing `a[ [ 1, 2 ] ]`, but I'm not.

## Actual behavior

When I use auto-correct, the two cops fight each other and create an infinite loop.

## Steps to reproduce the problem

Create the files above as `style.yml` and `a.rb`, then run:

```
rubocop -c style.yml a.rb  --auto-correct
```

## RuboCop version

0.52.0 (using Parser 2.4.0.2, running on ruby 2.2.2 x86_64-linux)

**Comment by garettarrowood** [comment]
This is an issue in `Layout/SpaceInsideReferenceBrackets` with right_bracket detection.  Thanks for the report. I'll take a look at this soon.

**Comment by aquach** [comment]
Cheers, thanks!

**Comment by shevegen** [comment]
Most epic line in this issue so far :)

**When I use auto-correct, the two cops fight each other and create an infinite loop.**

**Comment by aquach** [comment]
Thanks so much!
